plugins/statuslines/lualine.nix

```diff
@@ -36,7 +36,7 @@ with lib; let
               Enables the display of icons alongside the component.
             '';
 
-            icon = helpers.mkNullOrOption types.str ''
+            icon = helpers.mkNullOrOption (with types; either str (attrsOf anything)) ''
```

Comment:
```suggestion
            icon = helpers.mkNullOrOption
              (
                with types;
                  either
                  str
                  (submodule {
                    freeformType = attrsOf anything;
                    
                    options = {
                      icon = mkOption {
                        type = str;
                        description = "Icon character.";
                      };
                    };
                  })
              )
              "Defines the icon to be displayed in front of the component.";
```

Reply:
Good job ! I suggest this for the option definition.
It ensures that the `icon.icon` option is set with a string.

Reply:
That's for sure better, I have applied it, need to remove the extra line, but after my testing works as expected.